src/index.tsx

```diff
@@ -518,11 +518,46 @@ export function useEndpoints() {
   return endpoints;
 }
 
+/**
+ * This hook can set target track encoding.
+ */
+export function useTargetTrackEncoding() {
+  /**
+   * sets track encoding that server should send to the client library.
+   * The encoding will be sent whenever it is available. If choosen encoding is
+   * temporarily unavailable, some other encoding will be sent until choosen encoding
+   *  becomes active again.
+   *
+   * @param trackId id of a track which encoding you want to select
+   * @param encoding encoding to select
+   */
+  const setTargetTrackEncoding = useCallback(
```

Comment:
no need for a hook, just function is enough 
see for example `updateEndpointMetadata` function